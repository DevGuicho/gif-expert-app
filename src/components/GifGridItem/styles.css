.container__gif {
  position: relative;
  margin: 10px 0;
}
.gif {
  vertical-align: middle;
  width: 100%;
}
.gif__caption {
  position: absolute;
  bottom: 0;
  left: 0;
  color: white;
  padding: 3px;
  font-size: 12px;
  line-height: 18px;
  background-color: rgba(0, 0, 0, 0.5);
}
