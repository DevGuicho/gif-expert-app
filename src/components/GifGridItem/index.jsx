import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const GifGridItem = ({ url, title }) => {
  return (
    <figure className="container__gif animate__animated animate__fadeInDown">
      <img className="gif" src={url} alt={title} />
      <figcaption className="gif__caption">{title}</figcaption>
    </figure>
  );
};

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default GifGridItem;
