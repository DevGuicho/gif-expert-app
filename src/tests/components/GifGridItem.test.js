import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem url={url} title={title} />);

  test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe tener un parrafo con el titulo', () => {
    const figcaption = wrapper.find('figcaption');
    expect(figcaption.text().trim()).toBe(title);
  });

  test('Debe de tener la imagan igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener animate__fadeInDown', () => {
    const figure = wrapper.find('figure');
    expect(figure.prop('className').includes('animate__fadeInDown')).toBe(true);
  });
});
