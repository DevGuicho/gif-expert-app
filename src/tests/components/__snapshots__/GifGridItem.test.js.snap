// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<figure
  className="container__gif animate__animated animate__fadeInDown"
>
  <img
    alt="Un titulo"
    className="gif"
    src="https://localhost/algo.jpg"
  />
  <figcaption
    className="gif__caption"
  >
    Un titulo
  </figcaption>
</figure>
`;
